lang/julia/README.md

```diff
@@ -0,0 +1,371 @@
+# Julia
+
+In this example, we will cover:
+ - [Pulling a julia container from the library.](#then-pull-the-container-from-the-library)
+ - [Building the container from a recipe.](#to-build-from-a-recipe)
+ - [Running a julia script.](#to-run-a-julia-script)
+
+<br>
+
+Now of corse you can build the [Official julia container from docker](https://hub.docker.com/_/julia/), <br>
+But for this example we will install julia in a [Ubuntu container](https://hub.docker.com/_/ubuntu/). <br>
+
+<br>
+
+#### What you need:
+ - Singularity, which you can download and install from [here](https://github.com/sylabs/singularity).
+ - A text editor, like: `micro`, `vim` or `nano`.
+ - Root access (only if you're [building from a recipe](#to-build-from-a-recipe)).
+ 
+
+<br>
+
+____
+
+
+<br>
+
+
+#### To start, make the working directory:
+
+```
+$ mkdir julia
+$ cd julia/
+```
+
+<br>
+
+
+#### Then, pull the container from the library:
+
+```
+$ singularity pull library://sylabs/examples/julia.sif:latest
+```
+
+<br>
+
+#### Rename the container you pulled:
+
+```
+$ mv julia.sif_latest.sif julia.sif
+```
+
+<br>
+
+#### To run the container:
+
+```
+$ singularity run julia.sif 
+Hello world from: https://raw.githubusercontent.com/sylabs/examples/master/lang/julia/hello-world.jl
+For full tutorial, visit: https://github.com/sylabs/examples/lang/julia
+$
+```
+
+<br>
+<br>
+
+### To build from a recipe:
+
+You will need root access to build from a recipe.
+
+#### First, make the definition file: (aka, the recipe)
+
+```
+$ nano julia.def
+```
+```
+BootStrap: docker
+From: ubuntu:16.04
+
+%runscript
+# run your script here, eg.
+# julia testing.jl
+echo 'println("hello world from julia script!!!")' | julia
+
+%environment
+export PATH=/julia-1.0.1/bin:$PATH
+export LD_LIBRARY_PATH=/julia-1.0.1/lib:/julia-1.0.1/lib/julia:$LD_LIBRARY_PATH
+export LC_ALL=C
+
+%post
+apt-get -y update
+# we are installing some basic packages,
+# you can install your own
+#apt-get -y install <YOUR_PACKAGE>
+
+# install some basic tools
+apt-get -y install curl
+apt-get -y install tar
+apt-get -y install gzip
```

Comment:
Let's change the 3 lines above to:
```
apt-get -y install curl tar gzip
```
I much prefer that syntax for brevity.

Reply:
Done.